src/tools/perf/perftest.c

```diff
@@ -744,24 +728,27 @@ static ucs_status_t setup_mpi_rte(struct perftest_context *ctx)
     ctx->params.super.rte_group  = NULL;
     ctx->params.super.rte        = &mpi_rte;
     ctx->params.super.report_arg = ctx;
-#endif
     return UCS_OK;
 }
 
 static ucs_status_t cleanup_mpi_rte(struct perftest_context *ctx)
 {
-#if defined (HAVE_MPI)
     void *buffer;
     int size;
 
     MPI_Buffer_detach(&buffer, &size);
     ucs_assert(buffer != NULL);
     ucs_assertv(size == MPI_RTE_BSEND_BUFFER_SIZE, "size=%d", size);
     free(buffer);
-#endif
     return UCS_OK;
 }
 
+static const perftest_rte_t plugin_mpi_rte = {
```

Comment:
I think no, because we are adding perftest plugin that manipulates `struct perftest_context` which is perftest wide. The plugin implements accept connection, test params transfer, text reports generation. Also part of that is to setup `ucx_perf_rte_t`, but this is only libperf specific.

Reply:
now `struct perftest_context` could itself be properly wrapped with setup/cleanup/priv based implementation, but i'd keep it for later patch tbh.

Reply:
> I think no, because we are adding perftest plugin that manipulates `struct perftest_context` which is perftest wide. The plugin implements accept connection, test params transfer, text reports generation. Also part of that is to setup `ucx_perf_rte_t`, but this is only libperf specific.

I don't think it's a problem , the rte struct would just define the operations and be placed in a global list, and perftest context would point to the selected rte